backend/project-service/src/jvmMain/kotlin/project/rpc/ProjectController.kt

```diff
@@ -285,14 +286,32 @@ class ProjectController(
         }
 
         implement(Projects.rename) {
-            ok(
+            val requestedProject = queries.lookupById(db, request.id) ?: throw RPCException(
+                "No project with that id",
+                HttpStatusCode.BadRequest
+            )
+            if (requestedProject.parent != null && requestedProject.parent == ctx.project) {
+                val isAdmin = queries.isAdminOrPIOfProject(db, ctx.securityPrincipal.username, requestedProject.parent!!)
+                if (isAdmin) {
+                    projects.renameProject(
+                        db,
+                        ctx.securityPrincipal.toActor(),
+                        request.id,
+                        request.newTitle,
+                        subProject = true
+                    )
+                } else {
+                    throw RPCException.fromStatusCode(HttpStatusCode.Forbidden, "Not admin")
```

Comment:
I would change both exceptions in this block to a more generic `NotFound` exception.  That way we don't potentially leak information about if a project exists or not.

Reply:
```suggestion
                    throw RPCException.fromStatusCode(HttpStatusCode.NotFound)
```